Update shelly.markdown with clearer language and more actionable steps (#28487)
home-assistant/home-assistant.github.io: source/_integrations/shelly.markdown

## Shelly device generations

There are two generations of Shelly devices, and this integration supports both. There are some differences in how devices should be configured and in the naming of entities and devices between generations. Some of the ways to tell the Shelly generations apart are:

- In Home Assistant UI, on the Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`;
- 2nd generation devices will have a web UI sidebar on the left;

> - 2nd generation devices will have a blue Shelly favicon on their web browser tab.

Reviewer: In my opinion, a better way to distinguish the generation of devices is the model name, if the name includes Plus/Pro, it is a Gen2 device. Users probably know what device they bought, they should know that IMO.

Reviewer 2: We should not point to web colors, @Chef-de-IT current Shelly beta which is starting role out already changed the web colors from previous version due to major web re-design. The HA gen1/gen2 or Plus/Pro in the name is the correct way.

Author: > if the name includes Plus/Pro, it is a Gen2 device

[bieniu](https://github.com/bieniu) , in the live version of this doc last section https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/shelly.markdown#known-issues-and-limitations I'm reading:

> Generation 1 "Shelly 4Pro" and "Shelly Sense" are not supported (devices based on old CoAP v1 protocol)

I may be wrongly conflating "old protocol" with "it makes no sense for this to be the latter generation device". Is Shelly 4Pro such as it's mentioned in the quoted doc, generation 2, generation 1, or pre-generation 1 (say generation 0)?? 

Shelly doesn't print generation 1 vs 2 on the product carton or even among the specs they list on the website - to say nothing of the listing on Amazon. As I type this, I'm holding a _**Shelly** dimmer **2**_ box (Shelly and 2 are imprinted in bold), yet it's generation 1. Given all of this, how can we realistically say "user should know" ? Know how?

I feel, the point of a doc, especially a troubleshooting doc, is to help as broad a range of users in trouble as possible, to resolve their issues and go on having a good HA experience. The rest of us nerds can browse the code.

## Shelly device generations

> There are two generations of Shelly devices, and this integration supports both. In Home Assistant UI, on each Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`. There are differences in how gen1 vs. gen2 devices should be configured, and in the naming of entities and devices.

Reviewer: ```suggestion
There are two generations of Shelly devices, and this integration supports both. In Home Assistant UI, on each Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`. There are differences in how generation 1 vs. generation 2 devices should be configured, and in the naming of entities and devices.
```

> <div class="note">Shelly cloud connection is not required for this HA integration to work. Shelly devices are communicating directly with this HA integration.</div>

Reviewer: ```suggestion
<div class="note">Shelly cloud connection is not required for this integration to work. Shelly devices are communicating directly with Home Assistant server.</div>
```

We do not need to repeat that the case concerns "HA integration". Besides, I'm not a fan of abbreviations in the documentation, it's better to use the full name of Home Assistant.

## Configuring 1st generaton Shelly devices

> First generation Shelly devices use the `CoIoT` protocol to communicate with the HA integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:

Reviewer: ```suggestion
First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:
```

> First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. To fix:

Reviewer 3: ```suggestion
First generation Shelly devices use the `CoIoT` protocol to communicate with the integration. Home Assistant will display a repair issue for any such Shelly device, if push updates from it do not reach the Home Assistant server. Suggested steps to fix:
```

> 1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;

Reviewer 3: We should tell the minimum version, "current" is not enough: what if the device for whatever reason is not offering a new firmware for example ?

Author: Good idea & intention - I'm not aware if ALL devices share the same firmware version # nomenclature. The goal of this PR is to provide actionable helpful steps to as many people with this issue as possible, as soon as possible - also heeding the Home Assistant dev docs advice to keep PRs small and fast rather than make them the be-all end-all.

Author: Identified and added the vsersion

2. In the device's web UI (at the local IP address of your Shelly device), go to: **Internet & Security** >> **ADVANCED - DEVELOPER SETTINGS**;
3. Make sure box **Enable CoIoT** is selected;

> 4. In **CoIoT peer** field, enter `Your_Home_Assistant_local_IP_address:5683`

Reviewer 3: ```suggestion
4. In **CoIoT peer** field, enter `Your_Home_Assistant_local_IP_address:5683`, for example: `192.168.1.10:5683`

```

> 5. **Save**;

Reviewer: ```suggestion
5. Click **Save**;
```

6. Reboot the Shelly device.

> If you have followed the above steps, but the HA repair issue remains, there are diagnostic steps and ideas listed in the **Known issues and limitations** section at the bottom of this document.

Reviewer: ```suggestion
If you have followed the above steps, but the repair issue remains, there are diagnostic steps and ideas listed in the **Known issues and limitations** section at the bottom of this document.
```

## Configuring 2nd generation Shelly devices

> Second generation Shelly devices use the `RPC` protocol to communicate with the integration. _Battery powered gen. 2 devices **only**_ need manual outbound WebSocket configuration:

Reviewer: ```suggestion
Second generation Shelly devices use the `RPC` protocol to communicate with the integration. Battery powered 2nd generation devices **only** need manual outbound WebSocket configuration:
```

> Second generation Shelly devices use the `RPC` protocol to communicate with the integration. Battery powered 2nd generation devices **only** need manual outbound WebSocket configuration:

Reviewer 3: ```suggestion
Second generation Shelly devices use the `RPC` protocol to communicate with the integration. **Only** battery powered 2nd generation devices  need manual outbound WebSocket configuration:
```

> 1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;

Reviewer 3: Same as above, we should specify the minimum version of the firmware

Author: Please suggest the change to the text with the appropriate firmware version # (I reckon whatever was current mid-August this year that is known to have the input fields that resolve the HA issue in question)

Author: Identified and added the version, thanks

2. In the device's web UI (at the local IP address of your Shelly device), go to: Left sidebar >> **Settings** >> **Outbound WebSocket**;
3. Make sure box **Enable Outbound WebSocket** is selected;

> 4. In Server field, enter ws://`Your_Home_Assistant_local_IP_address:Port`/api/shelly/ws (note: port must match your HA port when you open HA UI);

Reviewer: ```suggestion
4. In Server field, enter `ws://Your_Home_Assistant_local_IP_address:Port/api/shelly/ws` (note: port must match your Home Assistant port when you open the UI);
```

> 4. In Server field, enter `ws://Your_Home_Assistant_local_IP_address:Port/api/shelly/ws` (note: port must match your Home Assistant port when you open the UI);

Reviewer 3: I would add an example here as well

Author: Earlier I was asked to DELETE a whole lot of specifics because "it was getting too long and technical". Let's try to navigate this down the middle to where it actually goes in and helps.

> 5. You *might* have to tweak the **Connection Type**, depending on your local TLS certificate situation;

Reviewer 3: I would add the most common one and then say "you may need to tweak...."

Author: I wouldn't say there's a common one - different folks run their networks differently. If somebody isn't deliberate with their network security there are only 3 options and they can try "whichever one works" but I'm loathe to put it in writing like that.

## Shelly device generations

There are two generations of Shelly devices, and this integration supports both. There are some differences in how devices should be configured and in the naming of entities and devices between generations. Some of the ways to tell the Shelly generations apart are:

- In Home Assistant UI, on the Shelly device's page, **Device info** section will state `Hardware: gen1` or `Hardware: gen2`;
- 2nd generation devices will have a web UI sidebar on the left;
- 2nd generation devices will have a blue Shelly favicon on their web browser tab.

> > [!NOTE]  

Reviewer: You should use `<div class="note"></div>` intead of `[!NOTE]`.

Author: I'll make these changes throughout the doc. Was initially under the impression that relying on GitHub's native alert box markdown ( https://github.com/orgs/community/discussions/16925 ) instead of styled divs was semantically cleaner, but will switch back to divs.

4. In **CoIoT peer** field, enter `Your_Home_Assistant_local_IP_address:5683`
5. **Save**;
6. Reboot the Shelly device.

If you have followed the above steps, but the HA repair issue remains, there are diagnostic steps and ideas listed in the **Known issues and limitations** section at the bottom of this document.

## Configuring 2nd generation Shelly devices

Second generation Shelly devices use the `RPC` protocol to communicate with the integration. **Battery powered gen. 2 devices only** need manual outbound WebSocket configuration:

1. Ensure the Shelly device is on the current firmware. Older firmware will lack the necessary setting;
2. In the device's web UI (at the local IP address of your Shelly device), go to: Left sidebar >> **Settings** >> **Outbound WebSocket**;
3. Make sure box **Enable Outbound WebSocket** is selected;
4. In Server field, enter ws://`Your_Home_Assistant_local_IP_address:Port`/api/shelly/ws (note: port must match your HA port when you open HA UI);
5. You *might* have to tweak the **Connection Type**, depending on your local TLS certificate situation;
6. Click **Apply** to save the settings;
7. Reboot the Shelly device.

> > [!NOTE]  

Reviewer: You should use `<div class="note"></div>` instead of `[!NOTE]`.

## Known issues and limitations

> - Generation 1 "Shelly 4Pro" and "Shelly Sense", *based on the old CoAP v1 protocol*, are **not** supported.

Reviewer: ```suggestion
- Generation 1 "Shelly 4Pro" and "Shelly Sense", based on the old CoAP v1 protocol, are not supported.
```

- Only supports firmware 1.8 and later for generation 1 devices
- Only supports firmware 0.8 and later for generation 2 devices
- Before set up, battery-powered devices must be woken up by pressing the button on the device.
- OTA update service does not support battery-powered devices
- Frequency sensors for generation 2 energy monitoring devices are supported by firmware 1.0.0 beta 6 or later.

### Troubleshooting first generation Shelly devices raising a HA repair issue

> The following ideas might help you diagnose and fix the problem of the gen. 1 Shelly device CoIoT push updates not propagating to HA, and causing HA to display a repair issue:

Reviewer: ```suggestion
The following ideas might help you diagnose and fix the problem of 1st generation Shelly device CoIoT push updates not propagating to Home Assistant:
```

> - Check that communication on `UDP` port `5683` is propagating from your Shelly device to Home Assistant. This can be affected when the HA is behind a firewall (such as one running on HA's virtual machine host) or in a different subnet or VLAN than Shelly devices. Connect a computer to the same WiFi SSID at the same location / access point as the affected Shelly device(s) and try:

Reviewer: ```suggestion
- Check that communication on `UDP` port `5683` is propagating from your Shelly device to Home Assistant. This can be affected when the Home Assistant server is behind a firewall (such as one running on the virtual machine host) or in a different subnet or VLAN than Shelly devices. Connect a computer to the same WiFi SSID at the same location/access point as the affected Shelly device(s) and try:
```

  - On Windows command line: `portqry.exe -n Your_Home_Assistant_local_IP_address -p udp -e 5683 -sp 5683` This tool is available at https://www.microsoft.com/en-us/download/details.aspx?id=17148 or its GUI version PortQryUI at https://www.microsoft.com/en-us/download/details.aspx?id=24009&wa=wsignin1.0

>   - On Linux: `nc -vz -u Your_Home_Assistant_local_IP_address 5683`

Reviewer: It seems to me that the user may not understand the output of these tools. Maybe an example is needed.

Author: Added in most recent commit: desired output identifiers and links to usage and other examples

- Check WiFi connection quality. In the device's web UI, go to **Settings** >> **DEVICE INFO**. Confirm the proper SSID name and signal strength; if device reports WiFi RSSI of -70 dBm or a greater negative value, signal strength may be the issue. You can test this hypothesis by temporarily relocating the affected Shelly device(s) an/or their WiFi access point(s) to be closer together and/or with fewer metal obstacles between them, so that you get the shelly device(s) report the WiFi RSSI in the -40s...low 50s dBm. If the issue goes away then, one or more of the following measures may help:
  - Optimize WiFi access point placement to boost the worst affected devices' Wi-Fi signal. Minimizing metal obstructions (includig wires) in the immediate vicinity of Wi-Fi antennae can often make a meaningful difference.
  - If your affected Shelly devices are installed in metal electrical boxes, **only if local electrical codes allow**, consider having a qualified electrician replace any unused metal electrical box knock-outs with NM (non-metallic) push-in connectors. This will allow  additional Wi-Fi signal pathways to the Shelly device.

>   - If your affected Shelly devices have a dense mass of electrical wiring surrounding them, consider having a qualified electrician clean up the wiring. Multi-conductor lever connectors, such as WAGO 221-415 *(example - not endorsement)*, can allow for more compact connections compared to traditional twist nuts.

Reviewer: I'm not sure we need this. This is HA integration documentation, not an electrical manual.

Author: This is an excellent point to ponder:

- On the one hand, the reality is,
  - many Shelly switches are situated in electrical boxes whose location and material are governed by local electrical codes, 
  - users do report very, very low signal strengths / poor connections, 
  - a very poor connection is something known to create problems.
  
This hurts the entire HA ecosystem by spoiling the user experience and adding support overhead - hence dampening HA's vitality and spread. Unmitigated poor connections (where it could be fixed with the right knowledge) belong in a lab. So whether a firewall, a subnet, a VLAN, or a metal electrical box are blocking the communication, the end result is the same. Hence, awareness ought to be the same. Not everyone has RF communications knowledge, and most electricians simply treat smart switches as just regular switches a customer provided. HA user awareness and directions at install time along the lines outlined in the PR can significantly affect the outcomes.
- On the other hand, the same RF propagation and signal strength issues can and do afflict not just Shelly but many other devices, such as Z-Wave, Zigbee, and a plethora of other wireless protocols. Hence:
  - While I believe firmly that this awareness and advice (expanding on basics already in the version currently live and offering additional options where, e.g. the current advice to relocate the electrical box with a Shelly device may be quite impractical for many installations),
  - I do agree with your point that this RF knowledge and solution ideas belongs somewhere other than Shelly integration documentation.

I'll therefore trim this section to a bare minimum and will look for a place to open a separate PR that will add this knowledge that can be linked from any integrations relying on wireless communication.

  - If you are using a metal wallplate and the WiFi signal improves with it removed, consider switching to plastic.
  - If using a Wi-Fi network with multiple access points, in the Shelly device's web UI, enable **Internet & Security** >> **WiFi Client AP Roaming** option.
  - Consider adding another WiFi access point, situated where it will improve the connection quality with the affected Shelly device(s).
- Check that your Shelly devices' time is consistent with your HA server and your router, with the following settings in Shelly device web UI:
  - **Settings** >> **TIME ZONE AND GEO-LOCATION** >> **Local Time zone of the Shelly device**
  - **Settings** >> **TIME ZONE AND GEO-LOCATION** >> **Daylight saving time**
  - **Internet & Security** >> **SNTP Server** (make sure this server is reachable)
- Consider **TEMPORARILY** enabling Debug Log in **Settings** >> **DEVICE INFO**, just long enough to capture any current errors the Shelly device is reporting, and then disabling the Debug Log.

> > [!WARNING]  

Reviewer: You should use `<div class="note warning"></div>`instead of `[!WARNING]`.

> Leaving the Debug Log enabled for an extended period will age the Shelly device's flash memory, i.e. use up its finite write endurance cycles.

> > [!NOTE]  

Reviewer: You should use `<div class="note"></div>` instead of `[!NOTE]`.
